kore/src/Kore/Rewrite/RewriteStep.hs

```diff
@@ -290,16 +294,36 @@ finalizeRulesParallel
                     unifiedRules
                     & fmap fold
             let unifications = MultiOr.make (Conditional.withoutTerm <$> unifiedRules)
-                remainder = Condition.fromPredicate (Remainder.remainder' unifications)
-            remainders <-
-                applyRemainder sideCondition initial remainder
-                    & Logic.observeAllT
-                    & fmap (fmap assertRemainderPattern >>> OrPattern.fromPatterns)
-            return
-                Step.Results
-                    { results = Seq.fromList results
-                    , remainders
-                    }
+                remainderPredicate = Remainder.remainder' unifications
+            -- evaluate the remainder predicate to make sure it is actually satisfiable
+            SMT.evalPredicate
+                (WarnDecidePredicateUnknown $srcLoc Nothing)
+                remainderPredicate
+                Nothing
+                >>= \case
+                    -- remainder condition is UNSAT: we prune the remainder branch early to avoid
+                    -- jumping into the pit of function evaluation in the configuration under the
+                    -- contradictory condition (the unsatisfiable remainder)
+                    Just False ->
+                        return
+                            Step.Results
+                                { results = Seq.fromList results
+                                , remainders = mempty
+                                }
+                    -- remainder condition is UNKNOWN: TODO: what happens here?
```

Comment:
With the current code for `evalPredicate`, the result will never be `Just True`.  When the solver returns `Unknown` the result is `Nothing` here, the same as for `Sat` from the solver.
We could instead stop execution by passing `ErrorDecidePredicateUnknown` above. If the precondition is already an `Unknown` problem for the solver we will likely hit that error later anyway.